scopecheck/scopecheck.go

```diff
@@ -0,0 +1,85 @@
+package scopecheck
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+)
+
+func init() {
+	extpoints.Register("scope-check", scopecheck{})
+}
+
+type scopecheck struct{}
+
+func (scopecheck) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (scopecheck) Summary() string {
+	return "Shows whether a given scope or set of scope satifies another."
+}
+
+func (scopecheck) Usage() string {
+	return `Usage:
+  taskcluster scope-check <scope1> satisfies <scope2>
+`
+}
+
+func expandScope(scope2 []string) (*auth.SetOfScopes, error) {
+
+	a := auth.New(&tcclient.Credentials{})
+	a.Authenticate = false
+
+	scopes := &auth.SetOfScopes{
+		Scopes: scope2,
+	}
+
+	resp, err := a.ExpandScopes(scopes)
+	if err != nil {
+		fmt.Printf("Error expanding scopes: %s\n", err)
+		return nil, err
+	}
+
+	return resp, err
+}
+
+func (scopecheck) Execute(context extpoints.Context) bool {
+	argv := context.Arguments
+	scope1 := argv["<scope1>"].(string)
+	if argv["satisfies"] == true {
```

Comment:
when I do `satisfies =: argv["satisfies"]` I am introducing a variable satisfies which the compiler will complain about because I wont use it anywhere later...